File: backend/main-service/src/main/java/main/service/ExerciseService.java
package main.service;

import main.dto.ExerciseDataDto;
import main.dto.ExerciseDto;
import main.dto.ExerciseResponse;

import java.util.List;

public interface ExerciseService {
    //For get all exercise with muscle name and pagination
    ExerciseResponse getByMuscleName(String name, Integer page);
    //For get exercise with id
    ExerciseDataDto getExercise(long exerciseID);
    //Searching with exercise or muscle name and pagination
    ExerciseResponse search(String name, Integer page);
    //Sliders
    List<ExerciseDto> armSlider();
    List<ExerciseDto> legSlider();
    List<ExerciseDto> chestSlider();
    //Is for fining some suggest names
    List<String> searchName(String name);
}

File: frontend/main/src/app/service/exercise-service.ts
import { ExerciseDataDto, ExerciseDto, ExerciseResponse } from "../types/exercise-type";

export const exerciseService = {
  async getExercise(name: string, page: number = 0) {
    const response = await fetch(`/api/exercise/by-muscle/${name}?page=${page}`, {
      method: 'GET'
    });
    if (!response.ok) throw new Error('Failed to fetch exercise');
    return response.json() as Promise<ExerciseResponse>;
  },

  async search(name: string, page: number = 0) {
    const response = await fetch(`/api/exercise/search/${name}?page=${page}`, {
      method: 'GET'
    });
    if (!response.ok) throw new Error('Failed to fetch exercise');
    return response.json() as Promise<ExerciseResponse>;
  },

  async armSlider() {
    const response = await fetch(`/api/exercise/arm-slider`, {
      method: 'GET'
    });
    if (!response.ok) throw new Error('Failed to fetch exercise');
    return response.json() as Promise<ExerciseDto[]>;
  },

  async legSlider() {
    const response = await fetch(`/api/exercise/leg-slider`, {
      method: 'GET'
    });
    if (!response.ok) throw new Error('Failed to fetch exercise');
    return response.json() as Promise<ExerciseDto[]>;
  },

  async chestSlider() {
    const response = await fetch(`/api/exercise/chest-slider`, {
      method: 'GET'
    });
    if (!response.ok) throw new Error('Failed to fetch exercise');
    return response.json() as Promise<ExerciseDto[]>;
  },

  async findExercise(id: number) {
    const response = await fetch(`/api/exercise/get/${id}`, {
      method: 'GET'
    });
    if (!response.ok) throw new Error('Failed to fetch exercise');
    return response.json() as Promise<ExerciseDataDto>;
  },
}